### ISSUE TITLE
Unexpected pixels in lesion and SC segmentations

### ISSUE THREAD

**Issue by Nilser3** [ivadomed/model_seg_ms_mp2rage#76]
In some subjects of the `basel-mp2rage` dataset there are pixels inconsistent with the segmentations 

# sub-P047 :

![point_47](https://github.com/ivadomed/model_seg_ms_mp2rage/assets/77469192/3a9f096f-b952-4bd5-b64e-754291b07130)

An outer pixel is observed at the coordinates (108, 115, 116)

### UNIT1 header:

<details><summary>sct_image -i sub-P047_UNIT1.nii.gz -header</summary>

```console

--
Spinal Cord Toolbox (6.0)

sct_image -i sub-P047_UNIT1.nii.gz -header
--

sizeof_hdr      348
data_type       UINT16
dim             [3, 240, 256, 176, 1, 1, 1, 1]
vox_units       mm
time_units      s
datatype        512
nbyper          2
bitpix          16
pixdim          [1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0]
vox_offset      352
cal_max         0.000000
cal_min         0.000000
scl_slope       1.000000
scl_inter       0.000000
phase_dim       0
freq_dim        0
slice_dim       0
slice_name      Unknown
slice_code      0
slice_start     0
slice_end       0
slice_duration  0.000000
toffset         0.000000
intent          Unknown
intent_code     0
intent_name
intent_p1       0.000000
intent_p2       0.000000
intent_p3       0.000000
qform_name      Scanner Anat
qform_code      1
qto_xyz:1       0.000000 0.000000 1.000000 -88.823051
qto_xyz:2       -1.000000 0.000000 0.000000 142.514359
qto_xyz:3       0.000000 -1.000000 0.000000 106.335777
qto_xyz:4       0.000000 0.000000 0.000000 1.000000
qform_xorient   Anterior-to-Posterior
qform_yorient   Superior-to-Inferior
qform_zorient   Left-to-Right
sform_name      Scanner Anat
sform_code      1
sto_xyz:1       -0.000000 -0.000000 1.000000 -88.823051
sto_xyz:2       -1.000000 -0.000000 -0.000000 142.514359
sto_xyz:3       0.000000 -1.000000 0.000000 106.335777
sto_xyz:4       0.000000 0.000000 0.000000 1.000000
sform_xorient   Anterior-to-Posterior
sform_yorient   Superior-to-Inferior
sform_zorient   Left-to-Right
file_type       NIFTI-1+
file_code       1
descrip         MRtrix version: 3.0_RC3-91-gd2cd716d
aux_file

```
</details>

### Lesion segmentation header (manualNeuroPoly mask) :

<details><summary>sct_image -i sub-P047_UNIT1_lesion-manualNeuroPoly.nii.gz -header</summary>

```console

--
Spinal Cord Toolbox (6.0)

sct_image -i sub-P047_UNIT1_lesion-manualNeuroPoly.nii.gz -header
--

sizeof_hdr      348
data_type       UINT16
dim             [3, 240, 256, 176, 1, 1, 1, 1]
vox_units       mm
time_units      Unknown
datatype        512
nbyper          2
bitpix          16
pixdim          [1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0]
vox_offset      352
cal_max         0.000000
cal_min         0.000000
scl_slope       1.000000
scl_inter       0.000000
phase_dim       0
freq_dim        0
slice_dim       0
slice_name      Unknown
slice_code      0
slice_start     0
slice_end       0
slice_duration  0.000000
toffset         0.000000
intent          Unknown
intent_code     0
intent_name
intent_p1       0.000000
intent_p2       0.000000
intent_p3       0.000000
qform_name      Scanner Anat
qform_code      1
qto_xyz:1       0.000000 0.000000 1.000000 -88.823051
qto_xyz:2       -1.000000 0.000000 0.000000 142.514359
qto_xyz:3       0.000000 -1.000000 0.000000 106.335777
qto_xyz:4       0.000000 0.000000 0.000000 1.000000
qform_xorient   Anterior-to-Posterior
qform_yorient   Superior-to-Inferior
qform_zorient   Left-to-Right
sform_name      Unknown
sform_code      0
sto_xyz:1       0.000000 0.000000 0.000000 0.000000
sto_xyz:2       0.000000 0.000000 0.000000 0.000000
sto_xyz:3       0.000000 0.000000 0.000000 0.000000
sto_xyz:4       0.000000 0.000000 0.000000 1.000000
sform_xorient   Unknown
sform_yorient   Unknown
sform_zorient   Unknown
file_type       NIFTI-1+
file_code       1
descrip
aux_file

```
</details>

### Lesion segmentation header (manualHaris mask) :

<details><summary>sct_image -i sub-P047_UNIT1_lesion-manualHaris.nii.gz -header</summary>

```console

--
Spinal Cord Toolbox (6.0)

sct_image -i sub-P047_UNIT1_lesion-manualHaris.nii.gz -header
--

sizeof_hdr      348
data_type       UINT16
dim             [3, 240, 256, 176, 1, 1, 1, 1]
vox_units       mm
time_units      Unknown
datatype        512
nbyper          2
bitpix          16
pixdim          [1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0]
vox_offset      352
cal_max         0.000000
cal_min         0.000000
scl_slope       1.000000
scl_inter       0.000000
phase_dim       0
freq_dim        0
slice_dim       0
slice_name      Unknown
slice_code      0
slice_start     0
slice_end       0
slice_duration  0.000000
toffset         0.000000
intent          Unknown
intent_code     0
intent_name
intent_p1       0.000000
intent_p2       0.000000
intent_p3       0.000000
qform_name      Scanner Anat
qform_code      1
qto_xyz:1       0.000000 0.000000 1.000000 -88.823051
qto_xyz:2       -1.000000 0.000000 0.000000 142.514359
qto_xyz:3       0.000000 -1.000000 0.000000 106.335777
qto_xyz:4       0.000000 0.000000 0.000000 1.000000
qform_xorient   Anterior-to-Posterior
qform_yorient   Superior-to-Inferior
qform_zorient   Left-to-Right
sform_name      Unknown
sform_code      0
sto_xyz:1       0.000000 0.000000 0.000000 0.000000
sto_xyz:2       0.000000 0.000000 0.000000 0.000000
sto_xyz:3       0.000000 0.000000 0.000000 0.000000
sto_xyz:4       0.000000 0.000000 0.000000 1.000000
sform_xorient   Unknown
sform_yorient   Unknown
sform_zorient   Unknown
file_type       NIFTI-1+
file_code       1
descrip
aux_file

```
</details>

# sub-P106 :

![point_106](https://github.com/ivadomed/model_seg_ms_mp2rage/assets/77469192/f0eb0dea-faff-4ba6-a5d8-0797a08c8a64)

Some pixels are observed on the outside of the SC in the coordinates: (112, 253, 81)

### UNIT1 header:

<details><summary>sct_image -i sub-P106_UNIT1.nii.gz -header</summary>

```console

--
Spinal Cord Toolbox (6.0)

sct_image -i sub-P106_UNIT1.nii.gz -header
--

sizeof_hdr      348
data_type       UINT16
dim             [3, 240, 256, 176, 1, 1, 1, 1]
vox_units       mm
time_units      s
datatype        512
nbyper          2
bitpix          16
pixdim          [1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0]
vox_offset      352
cal_max         0.000000
cal_min         0.000000
scl_slope       1.000000
scl_inter       0.000000
phase_dim       0
freq_dim        0
slice_dim       0
slice_name      Unknown
slice_code      0
slice_start     0
slice_end       0
slice_duration  0.000000
toffset         0.000000
intent          Unknown
intent_code     0
intent_name
intent_p1       0.000000
intent_p2       0.000000
intent_p3       0.000000
qform_name      Scanner Anat
qform_code      1
qto_xyz:1       0.000000 0.000000 1.000000 -88.704819
qto_xyz:2       -1.000000 0.000000 0.000000 144.698792
qto_xyz:3       0.000000 -1.000000 0.000000 86.433739
qto_xyz:4       0.000000 0.000000 0.000000 1.000000
qform_xorient   Anterior-to-Posterior
qform_yorient   Superior-to-Inferior
qform_zorient   Left-to-Right
sform_name      Scanner Anat
sform_code      1
sto_xyz:1       -0.000000 -0.000000 1.000000 -88.704819
sto_xyz:2       -1.000000 -0.000000 -0.000000 144.698792
sto_xyz:3       0.000000 -1.000000 0.000000 86.433739
sto_xyz:4       0.000000 0.000000 0.000000 1.000000
sform_xorient   Anterior-to-Posterior
sform_yorient   Superior-to-Inferior
sform_zorient   Left-to-Right
file_type       NIFTI-1+
file_code       1
descrip         MRtrix version: 3.0_RC3-212-gf2ed89d8
aux_file


```
</details>

### SC segmentation header:

<details><summary>sct_image -i sub-P106_UNIT1_label-SC_seg.nii.gz -header</summary>

```console

--
Spinal Cord Toolbox (6.0)

sct_image -i sub-P106_UNIT1_label-SC_seg.nii.gz -header
--

sizeof_hdr      348
data_type       UINT16
dim             [3, 240, 256, 176, 1, 1, 1, 1]
vox_units       mm
time_units      Unknown
datatype        512
nbyper          2
bitpix          16
pixdim          [1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0]
vox_offset      352
cal_max         0.000000
cal_min         0.000000
scl_slope       1.000000
scl_inter       0.000000
phase_dim       0
freq_dim        0
slice_dim       0
slice_name      Unknown
slice_code      0
slice_start     0
slice_end       0
slice_duration  0.000000
toffset         0.000000
intent          Unknown
intent_code     0
intent_name
intent_p1       0.000000
intent_p2       0.000000
intent_p3       0.000000
qform_name      Scanner Anat
qform_code      1
qto_xyz:1       0.000000 0.000000 1.000000 -88.704819
qto_xyz:2       -1.000000 0.000000 0.000000 144.698792
qto_xyz:3       0.000000 -1.000000 0.000000 86.433739
qto_xyz:4       0.000000 0.000000 0.000000 1.000000
qform_xorient   Anterior-to-Posterior
qform_yorient   Superior-to-Inferior
qform_zorient   Left-to-Right
sform_name      Unknown
sform_code      0
sto_xyz:1       0.000000 0.000000 0.000000 0.000000
sto_xyz:2       0.000000 0.000000 0.000000 0.000000
sto_xyz:3       0.000000 0.000000 0.000000 0.000000
sto_xyz:4       0.000000 0.000000 0.000000 1.000000
sform_xorient   Unknown
sform_yorient   Unknown
sform_zorient   Unknown
file_type       NIFTI-1+
file_code       1
descrip
aux_file


```
</details>

**Comment by jcohenadad** [comment]
Before investigating this issue, I'd like to clarify #77

**Comment by Nilser3** [comment]
Fixed and merged to the `master` in [#265](https://github.com/neuropoly/data-management/issues/265)
